Makefile

```diff
@@ -436,12 +437,8 @@ CXXFLAGS += -pg
 LINKFLAGS += -pg
 endif
 
-ifeq ($(ENABLE_NDEBUG),1)
-CXXFLAGS := -O3 -DNDEBUG $(filter-out -Os -ggdb,$(CXXFLAGS))
-endif
-
 ifeq ($(ENABLE_DEBUG),1)
-CXXFLAGS := -Og -DDEBUG $(filter-out -Os,$(CXXFLAGS))
+CXXFLAGS := -Og -DDEBUG $(filter-out $(OPT_LEVEL),$(CXXFLAGS))
```

Comment:
The whole `$(filter-out $(some_flag),$(CXXFLAGS))` thing is ridiculous and ought to be gone.

Reply:
In this case it is better to override OPT_LEVEL with -Og value then to filter-out